presto-main/src/test/java/com/facebook/presto/execution/TestMemoryRevokingScheduler.java

```diff
@@ -166,13 +171,20 @@ public void testScheduleMemoryRevoking()
         OperatorContext operatorContext5 = driverContext211.addOperatorContext(5, new PlanNodeId("na"), "na");
 
         List<SqlTask> tasks = ImmutableList.of(sqlTask1, sqlTask2);
-        MemoryRevokingScheduler scheduler = new MemoryRevokingScheduler(singletonList(memoryPool), () -> tasks, executor, 1.0, 1.0, ORDER_BY_CREATE_TIME);
-
+        MemoryRevokingScheduler scheduler = new MemoryRevokingScheduler(
```

Comment:
Missing `MemoryRevokingScheduler#stop()` which would now leak the single threaded executor until GC finalization. Minor, but worth avoiding. Maybe a `withMemoryRevokingScheduler(<params>, Consumer<MemoryRevokingScheduler>)` helper could help with the boiler-plate of `registerPoolListeners()` and `stop()` but would be a problem with the way that `InterruptedException` is thrown from `awaitAsynchronousCallbacksRun()`...

Open to your thoughts about the best way to resolve the tension between test sanity and ensuring the revoking executor remains single threaded even with future refactoring changes.

Reply:
ended up going with a simple try/finally pattern in the tests to start and stop the memory revoking scheduler